### PR TITLE
feat(payments-stripe): create doesPlanMatchSubplatInterval

### PR DESCRIPTION
**Pull request by julianpoy**
## Because

* We need to actually check interval_count rather than just interval

## This pull request

* Adds a handy util for doing so, and uses it within the stripe manager

## Issue that this pull request solves

Closes FXA-9536

### CODE REVIEW

File: libs/payments/stripe/src/lib/util/doesPlanMatchSubplatInterval.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  },
+  [SubplatInterval.HalfYearly]: {
+    interval: 'month',
+    intervalCount: 6,
+  },
+  [SubplatInterval.Yearly]: {
+    interval: 'year',
+    intervalCount: 1,
+  },
+} satisfies Record<SubplatInterval, Interval>;
```
Comment (StaberindeZA) [inline libs/payments/stripe/src/lib/util/doesPlanMatchSubplatInterval.ts:30]: Oh nice!

**Review by StaberindeZA — APPROVED** [review]
r+

File: libs/payments/stripe/src/lib/util/doesPlanMatchSubplatInterval.spec.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { StripePlanFactory } from '../factories/plan.factory';
```
Comment (pdehaan) [inline libs/payments/stripe/src/lib/util/doesPlanMatchSubplatInterval.spec.ts:1]:
```suggestion
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import { StripePlanFactory } from '../factories/plan.factory';
```

File: libs/payments/stripe/src/lib/util/doesPlanMatchSubplatInterval.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { StripePlan } from '../stripe.client.types';
```
Comment (pdehaan) [inline libs/payments/stripe/src/lib/util/doesPlanMatchSubplatInterval.ts:1]:
```suggestion
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import { StripePlan } from '../stripe.client.types';
```

**Pull request merged**